src/ppx/grfx/grfx_scope.cpp

```diff
@@ -140,5 +184,16 @@ Result ScopeDestroyer::AddObject(grfx::Queue* pParent, grfx::CommandBuffer* pObj
     return ppx::SUCCESS;
 }
 
+void ScopeDestroyer::ReleaseAll()
+{
```

Comment:
do you need to destroy the elems before clearing?

Reply:
This function is to release all the objects without destroying them. For cases where you're using scope destroyer if there's an error but otherwise you'd keep the objects.  There's a comment in the header file that explains it.

Reply:
if this is the case, maybe better name it `ReleaseAllWithoutDestroy()`?

Reply:
The idiom comes from `unique_ptr` and a few other pointer classes that use release vs reset. I think in general release means to release ownership of and don't do anything else. 

https://en.cppreference.com/w/cpp/memory/unique_ptr/release

But if you really think the longer function name will help, I'll change it.

Reply:
maybe because i am coming from game engine world, when i see Release, my first impression is releasing/destroying the resource (or at least decr ref count which might trigger a destroy), if i don't see this, I would worry: "hmm, there might be a a mem leak?", and check the code implementations to see if there is a bug :p But if the function is named *WithoutDestroy, at least I would know it is intentional to not destroy the elems here :)

But I see your point, and did a quick search in BigWheels, it seems that this is the only place to have a Release function for now, so I think as long as future Release function in BigWheels follows the same rule, it would be fine. 

So I am ok with both solutions :)

Reply:
Understood. Thanks. If it gets too confusing later we'll change it to the more explicit name.